java/opendcs/src/main/java/opendcs/dai/LocationLevelDAI.java

```diff
@@ -0,0 +1,89 @@
+package opendcs.dai;
+
+import decodes.tsdb.DbIoException;
+import opendcs.dao.LocationLevelDAO.LocationLevelValue;
+import opendcs.dao.LocationLevelDAO.LocationLevelSpec;
+import org.opendcs.database.api.DataTransaction;
+import org.opendcs.database.api.OpenDcsDataException;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Abstract interface for Location Level Data Access Interface.
+ * This interface defines the core operations for reading location level data,
+ * which can be implemented for different database types (CWMS, HDB, OpenDCS, etc.)
+ */
+public interface LocationLevelDAI extends DaiBase
+{
+    /**
+     * Get the latest location level value for a given location level
+     * @param locationLevelId The location level identifier (format varies by database)
+     * @return The latest LocationLevelValue or null if none found
+     * @throws DbIoException on database error
+     */
+    LocationLevelValue getLatestLocationLevelValue(String locationLevelId) 
+        throws DbIoException;
+    
+    /**
+     * Get the latest location level value with unit conversion
+     * @param locationLevelId The location level identifier
+     * @param targetUnits The desired units for the value (e.g., "ft", "m", "in")
+     * @param sourceUnits The source units to filter by (if null, no filter applied)
+     * @return The latest LocationLevelValue or null if none found
+     * @throws DbIoException on database error
+     */
+    LocationLevelValue getLatestLocationLevelValue(String locationLevelId, 
```

Comment:
Why do the store units need to be known here? We should just let user request in the units desired and throw an appropriate exception if the units don't apply.